lib/livebook/session.ex

```diff
@@ -1102,6 +1156,51 @@ defmodule Livebook.Session do
     {:noreply, state}
   end
 
+  def handle_cast({:set_app_settings, client_pid, app_settings}, state) do
+    client_id = client_id(state, client_pid)
+    operation = {:set_app_settings, client_id, app_settings}
+    {:noreply, handle_operation(state, operation)}
+  end
+
+  def handle_cast({:deploy_app, _client_pid}, state) do
+    state =
+      if Notebook.AppSettings.valid?(state.data.notebook.app_settings) do
```

Comment:
Why do we need to check for the validity here? Shouldn't we assume it is valid?

Reply:
Technically the app settings can be invalid. That's how they are initially for example, and even though we shouldn't get a deploy request in such case, I think double-checking makes sense. Also, we may want to add something like "Clear settings", to remove all configuration (and no longer store it in the notebook), so the settings would become invalid and this way we would avoid a race condition. Does that make sense?

Reply:
I thought that could be the case. Can we add a small code comments explaining this decision? Then ship it!